### PR TITLE
New pre-commit hook for PHP, JS and SCSS

### PR DESCRIPTION
**Pull request by claudiosanches**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Replaced our old pre-commit hook to use Husky, it's the same used in `wc-admin`, also using some linters like `wc-admin`.

### How to test the changes in this Pull Request:

1. Pull this branch
2. Checkout to a new branch to do tests `git checkout -b test/husky`
3. Do changes in some files (inserting bugs :smile:, try .SCSS, .JS and .PHP)
4. Run `npm install`
5. Try commit something

Note: if updated `composer` dependencies, you may need to install the pre-commit hook again, so run `rm .git/hooks/pre-commit && node ./node_modules/husky/husky.js install`

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -13,22 +13,9 @@
     "apigen/apigen": "^4",
     "nette/utils": "~2.3.0",
     "phpunit/phpunit": "6.*",
-    "woocommerce/woocommerce-git-hooks": "*",
```
Comment (rodrigoprimo) [inline composer.json]: Should we archive the `woocommerce/woocommerce-git-hooks` repo?

File: package.json [hunk 2]
```diff
@@ -61,5 +64,30 @@
...
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "linters": {
+      "*.php": [
+        "php -d display_errors=1 -l",
+        "composer run-script phpcs"
```
Comment (rodrigoprimo) [inline package.json]: `composer run-script phpcs` calls PHPCS using the following command `phpcs -s -p` whereas `woocommerce-git-hooks` uses `phpcs --encoding=utf-8 -n -p`. `--enconding=utf8-8` is not necessary as it is the default value, but maybe we should keep `-n` to keep ignoring warnings and to perform the same check in the pre-commit use as the check we perform on Travis?

File: package.json [hunk 3]
```diff
@@ -61,5 +64,30 @@
...
+      ],
+      "*.scss": [
+        "stylelint --syntax=scss --fix",
+        "git add"
```
Comment (rodrigoprimo) [inline package.json]: Could you please explain why `git add` is necessary here and in the block below?

**Comment by rodrigoprimo** [comment]
Besides what I mentioned above, could you also fix the conflicts?

**Comment by rodrigoprimo** [comment]
It seems that it is necessary to manually remove the old `.git/hooks/pre-commit` before running `npm install`, otherwise husky won't touch the pre-commit file.

> Note: if updated composer dependencies, you may need to install the pre-commit hook again, so run node ./node_modules/husky/husky.js install

Not sure if I follow. Could you please elaborate on why it might be necessary to install the pre-commit hook again after updating composer dependencies?

**Comment by rodrigoprimo** [comment]
I wonder if there is a way to call `node ./node_modules/husky/husky.js install` whenever `npm install` is called so that we make sure the pre-commit hook is always installed? I'm thinking something similar to what `woocommerce-git-hooks` does using Composer command events (https://github.com/woocommerce/woocommerce-git-hooks#usage)

File: composer.json (hunk 1 above)
Comment (claudiosanches) [reply composer.json]: No need, people may use it.

File: package.json (hunk 3 above)
Comment (claudiosanches) [reply package.json]: `stylelint --fix` will make changes again in the files, so we need to use `git add` in order to include all in the same commit.

**Comment by claudiosanches** [comment]
@rodrigoprimo 

> I wonder if there is a way to call node ./node_modules/husky/husky.js install whenever npm install is called so that we make sure the pre-commit hook is always installed?

Probably you missed it:

> Note: if updated composer dependencies, you may need to install the pre-commit hook again, so run node ./node_modules/husky/husky.js install

Necessary because we are removing `woocommerce/woocommerce-git-hooks` and the code responsible to remove it. But if you start a new clone or clean code, you don't need to run manually anything.
Clear now?

**Comment by claudiosanches** [comment]
@rodrigoprimo code updated, fixed conflicts and included step to remove `.git/hooks/pre-commit` if updating.

**Review by rodrigoprimo — APPROVED** [review]
Thanks for the changes and clarifications.

**Comment by rodrigoprimo** [comment]
Approved after your changes but leaving unmerged. Can you handle merging and notifying everyone to update their environments?

**Pull request merged**